src/core/ready.js

```diff
@@ -49,11 +47,28 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !++readyFiring ) {
+				while ( readyCallbacks.length ) {
+					fn = readyCallbacks.shift();
+					if ( jQuery.isFunction( fn ) ) {
+						fn.call( document, jQuery );
+					}
+				}
+			}
+
+			readyFiring--;
+		};
+
+		whenReady();
 	}
 } );
 
+// Make jQuery.ready promise compatible (gh-1778)
+jQuery.ready.then = jQuery.fn.ready;
```

Comment:
Oh, blurp, I think is a misunderstanding of `Promise.resolve(thenable)`. `Promise.resolve(thenable)` follows the thenable and eventually defers to its state. This means that the thenable is responsible for calling `.then`. Notice that your example doesn't resolve even when you don't `return this`.